### ISSUE TITLE
Incorrect target name causes issues with code editor and goals

### ISSUE THREAD

**Issue by darundal** [FreeCodeCamp/FreeCodeCamp#9082]
#### Challenge Name

<a href="https://www.freecodecamp.com/challenges/target-a-specific-child-of-an-element-using-jquery">Target a specific child of an element using jQuery</a>

#### Issue Description

If the target is incorrectly listed as ("target-nth-child(2)") as opposed to the correct ("target:nth-child(2)") then when the run tests button is clicked the code editor section will begin flash black/white and then settle on white. If you attempt to reload the page the editor will appear as it should however it will not display the goals until the correct target name is entered and you hit the run tests button.

#### Browser Information

* User Agent String:
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2486.0 Safari/537.36 Edge/13.10586
* Browser Name, Version: 
Microsoft Edge, 25.10586.0.0
* Operating System: 
Windows 10 Pro
* Mobile, Desktop, or Tablet: 
Desktop

#### Your Code

```js
<script>
  $(document).ready(function() {
    $("#target1").css("color", "red");
    $("#target1").prop("disabled", true);
    $("#target4").remove();
    $("#target2").appendTo("#right-well");
    $("#target5").clone().appendTo("#left-well");
    $("#target1").parent().css("background-color", "red");
    $("#right-well").children().css("color", "orange");
    $("#left-well").children().css("color", "green");
    $(".target-nth-child(2)").addClass("animated bounce");

  });
</script>

<!-- Only change code above this line. -->

<div class="container-fluid">
  <h3 class="text-primary text-center">jQuery Playground</h3>
  <div class="row">
    <div class="col-xs-6">
      <h4>#left-well</h4>
      <div class="well" id="left-well">
        <button class="btn btn-default target" id="target1">#target1</button>
        <button class="btn btn-default target" id="target2">#target2</button>
        <button class="btn btn-default target" id="target3">#target3</button>
      </div>
    </div>
    <div class="col-xs-6">
      <h4>#right-well</h4>
      <div class="well" id="right-well">
        <button class="btn btn-default target" id="target4">#target4</button>
        <button class="btn btn-default target" id="target5">#target5</button>
        <button class="btn btn-default target" id="target6">#target6</button>
      </div>
    </div>
  </div>
```

**Comment by squanto** [comment]
unable to reproduce on chrome and firefox for mac

**Comment by Bouncey** [comment]
I'm on Ubuntu with Chrome. When I use `target-nth-child(2)` and press Run Tests:

The run tests button stays in the `focus` state. 
The test icons do not update.

When I click elsewhere in the screen the run tests button loses its focus state.

When I change the code to `target:nth-child(2)`:

the iphone screen shows `Error: Syntax error, unrecognized expression: .target-nth-child(2)`

When I then press run tests, I get the success modal.

**Comment by Pasayadaan** [comment]
Observed the exact same results as mentioned by @Bouncey on Windows 10, Chrome.

**Comment by erictleung** [comment]
When I input your code and run tests, the tests disappear for me. I'm running Chrome on Mac OS X Yosemite.

![image](https://cloud.githubusercontent.com/assets/2754821/16553092/ffb30a44-417a-11e6-9c13-c25f2dd6c651.png)

Not sure how we would tackle this. The code is incorrect and it's creating unintended effects in the emulator.

cc/ @FreeCodeCamp/issue-moderators

**Comment by Bouncey** [comment]
Would be fixed to a point with this PR #9490 

There is a bigger bug at play, but we haven't found it yet
